airbyte-integrations/bases/base-normalization/normalization/transform_catalog/stream_processor.py

```diff
@@ -574,7 +596,40 @@ def cast_property_type(self, property_name: str, column_name: str, jinja_column:
                 return f'nullif(cast({column_name} as {sql_type}), "") as {column_name}'
             replace_operation = jinja_call(f"empty_string_to_null({jinja_column})")
             return f"cast({replace_operation} as {sql_type}) as {column_name}"
-        elif is_string(definition["type"]):
+        elif (data_type.REF_TYPE_VAR_NAME in definition and is_binary_datatype(definition[data_type.REF_TYPE_VAR_NAME])) or (
+            data_type.ONE_OF_VAR_NAME in definition and is_binary_datatype(definition)
+        ):
+            if self.destination_type.value == DestinationType.POSTGRES.value:
+                # sql_type = "bytea"
+                sql_type = jinja_call("type_binary()")
+                return f"cast(decode({column_name}, 'base64') as {sql_type}) as {column_name}"
```

Comment:
maybe worth defining a `decode_base64` macro, to move the destination-specific logic out of this file. Then if we make the `type_binary()` macro return `dbt_utils.type_string()` for destinations without a binary type, then we don't need so much destination-specific logic here.

I.e. we could do
```python
decode_call = jinja_call(f"decode_base64({column_name})")
return f"cast({decode_call} as {sql_type}) as {column_name}"
```
for all destination types